src/internal/components/ScaledTextInput.tsx

```diff
@@ -0,0 +1,53 @@
+import { useEffect, useRef, useState } from 'react';
+import { cn } from '../../styles/theme';
+import type { TextInputReact } from './TextInput';
+import { TextInput } from './TextInput';
+
+export function ScaledTextInput(props: TextInputReact) {
```

Comment:
Can you add some comments that explain when to use `ScaledTextInput` vs `TextInput`?